odoo/models.py

```diff
@@ -5726,6 +5726,14 @@ def exists(self) -> Self:
         new_ids, ids = partition(lambda i: isinstance(i, NewId), self._ids)
         if not ids:
             return self
+
+        # Special version without query
+        field_to_check = next((f for f in self._fields.values() if f.prefetch is True), None)
+        if field_to_check:
+            field_cache = self.env.cache._get_field_cache(self, field_to_check)
+            if all(_id in field_cache for _id in ids):
```

Comment:
I remember when `write` we cannot promise the `record` exist, but just update the field cache.
So `_id` in the `field_cache` here may be caused by `write`.
But maybe it is acceptable here?

Reply:
That's a good point.

Reply:
Nice catch, I didn't think about it. Thanks

> But maybe it is acceptable here?

IMO, No it is not. I cancel it.